src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1986,13 +2004,18 @@ AllDistributedRelationsInListColocated(List *relationList)
 
 		if (IsCitusTableType(relationId, APPEND_DISTRIBUTED))
 		{
-			/*
-			 * If we got to this point, it means there are multiple distributed
-			 * relations and at least one of them is append-distributed. Since
-			 * we do not consider append-distributed tables to be co-located,
-			 * we can immediately return false.
-			 */
-			return false;
+			if (gotAppendTable)
```

Comment:
I cannot follow this logic. Why do we decide to allow two append tables to be colocated now?

Reply:
As before, we say that two append tables are not-colocated. However before, we were saying that the query references non-colocated append tables even if it only references a single append table and this PR fixes that. This wasn't a problem before but now we're calling this function in more places, hence needed to fix that.

Reply:
> As before, we say that two append tables are not-colocated

 the code doesn't seem to do that anymore? Before this patch, it was impossible for any two append tables to be decided as colocated, now it is possible?

> non-colocated append

append tables are always non-colocated, so I cannot follow this

> if it only references a single append table

Well, then we should make that more generic, at the top of the function. If there is a single table, we always return true. Having a special case for append is confusing 

> SELECT key, value FROM append_table a WHERE key IN (SELECT key FROM append_table WHERE value > 100) ORDER BY 1,2;

This query now starts to work. Is this code change allowing it? Can you share the explain plan of this query?

Reply:
> I cannot follow this logic. Why do we decide to allow two append tables to be colocated now?

As we've discussed, this change has been reverted.